### ISSUE TITLE
[Post Mortem] eFolder Hanging on to DB Connections

### ISSUE THREAD

**Issue by mdbenjam** [department-of-veterans-affairs/caseflow-efolder#720]
# Summary
On November 16 at 12:40pm EST we got a PagerDuty alert saying "Average DatabaseConnections GreaterThanThreshold 200.0 for DBInstanceIdentifier dsva-appeals-efolder-prod." 
![image](https://user-images.githubusercontent.com/3885236/32976623-9face8d2-cbe8-11e7-9451-8f95d98d3a78.png)

We used New Relic to determine that there was likely no user impact:
![image](https://user-images.githubusercontent.com/3885236/32976633-f0782d26-cbe8-11e7-82a6-56c04118858f.png)

After more investigation we determined that the Reader rollout had likely increased traffic enough for us to trip this DB connection alert threshold. eFolder saw an increase in throughput as more users were accessing the site:

![image](https://user-images.githubusercontent.com/3885236/32976637-1364fc38-cbe9-11e7-95f4-80bd419e6fe9.png)
![image](https://user-images.githubusercontent.com/3885236/32976639-25ac4fcc-cbe9-11e7-8634-4c9a05ba2e6a.png)

We also noticed that the number of DB connections monotonically increases throughout the day, and only decreases when we scale down at night.

We decided that since this was likely due to traffic by new users and that our alerting threshold was too low (we can handle up to 1200 DB connections) that this was not an immediate problem.

# Timeline
12:40 PM EST Alert comes in.
12:44 PM EST Alexis pings Sunil and Nick to look into the issue.
12:51 PM EST Sunil and Nick determine the issue is not user facing.
~1:00 PM EST Sunil inquires about new users and suggests that we're seeing the increase because of the rollout.
1:15 PM EST Sunil hands off investigation to Mark and Anya.
1:52 PM EST Mark posts graphs showing an increase in page views that correlates with an increase in eFolder throughput.
1:55 PM EST Mark determines there is not an immediate problem that needs an urgent fix.
2:20 PM EST Lowell notes that the DB connections monotonically increase until we scale down each night.

# Action Items
[1) Investigate releasing DB connections when we no longer need them.](https://github.com/department-of-veterans-affairs/caseflow-efolder/issues/738)
[2) Investigate why we require so many concurrent DB connections.](https://github.com/department-of-veterans-affairs/caseflow-efolder/issues/739)

**Comment by kierachell** [comment]
- how did we determine that we can handle 1200 DB connections?
- how did we determine that the issue had no effect on the users? did anyone log into Prod and smoke test the application?
- once this issue becomes user facing: do we have the technical knowledge, tooling, ability to handle this issue quickly?

**Comment by kierachell** [comment]
This job is not helping either:
https://github.com/department-of-veterans-affairs/caseflow-efolder/issues/734

**Comment by mdbenjam** [comment]
1) 1200 connections was the number quoted by Alan or Sunil, I am not sure how that is determined.
2) We used a combination of Apdex score, throughput, and lack of support tickets or sentry errors to determine there was no user impact.
3) That's why we have the action items. This post mortem explains why we triggered an alert, we have investigations to do to reduce the number of DB connections we use.
